src/main/java/com/github/dbmdz/solrocr/solr/SolrOcrHighlighter.java

```diff
@@ -52,16 +52,21 @@
 public class SolrOcrHighlighter extends UnifiedSolrHighlighter {
   private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
 
-  private final Executor hlThreadPool;
+  private final Executor hlExecutor;
+  private final int readerSectionSize;
+  private final int readerMaxCacheEntries;
 
   public SolrOcrHighlighter() {
-    this(Runtime.getRuntime().availableProcessors(), 8);
+    this(Runtime.getRuntime().availableProcessors(), 8, 8 * 1024, 64 * 1024);
   }
 
-  public SolrOcrHighlighter(int numHlThreads, int maxQueuedPerThread) {
+  public SolrOcrHighlighter(
+      int numHlThreads, int maxQueuedPerThread, int readerSectionSize, int readerMaxCacheEntries) {
```

Comment:
I think `readerMaxCacheEntries` is currently actually `maxSectionCacheSize`. Do we have to divide by `readerSectionSize` to get the maximum number of cache entries?

Reply:
😱 omg, yes 😱 thank you